### PR TITLE
IGAPP-604: Single page endpoints

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web/.idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -2,20 +2,42 @@
   <code_scheme name="Project" version="173">
```
Comment (klinzo) [inline web/.idea/codeStyles/Project.xml:2]: Code style imported from prettier, if you don't like it I could revert it.

**Review by Taggotty — APPROVED** [review]
The tests look much cleaner than before. 👍🏼

File: api-client/src/endpoints/createCategoryParentsEndpoint.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      }
+      return `${baseUrl}/${city}/${language}/wp-json/extensions/v3/parents?&url=${cityContentPath}`
+    })
+    .withMapper((json: Array<JsonCategoryType>, params: ParamsType): Array<CategoryModel> => {
+      const basePath = `/${params.city}/${params.language}`
+
+      const parents = json.map(category => mapCategoryJson(category, basePath))
+
+      parents.push(
+        new CategoryModel({
```
Comment (maxammann) [inline api-client/src/endpoints/createCategoryParentsEndpoint.js:30]: I'm not sure whether it is wise to craft these weird CaregoryModels. We could also just make this endpoint return an array of paths. What do you think?

**Review by maxammann — APPROVED** [review]

File: api-client/src/endpoints/createCategoryParentsEndpoint.js (hunk 2 above)
Comment (klinzo) [reply api-client/src/endpoints/createCategoryParentsEndpoint.js:30]: We'd need at least the name and the path though to show breadcrumbs. We could do that though,yes. Or we just don't include the root at all. Not sure about that. On the other hand this just follows the behavior of the already existing categories endpoint, which is not really optimal though. So not sure.

Comment (klinzo) [reply api-client/src/endpoints/createCategoryParentsEndpoint.js:30]: I think I'd leave this for now, we can refactor this at a later point of time together with the other endpoints using these weird models.

**Pull request merged**